### PR TITLE
[PREVIEW] Add uploadFailureCount column

### PR DESCRIPTION
**Pull request by rkondratowicz**
Only used for reads in this one.
Incrementing the value after failure will be done in next PR.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepository.java [hunk 1]
```diff
@@ -45,16 +45,21 @@
...
-     * @param status to filter upon
      * @param pageable limit of data to be processed by consumer
      * @return A list of envelopes
      */
-    List<Envelope> findByJurisdictionAndStatusOrderByCreatedAtAsc(
-        String jurisdiction,
-        Status status,
+    @Query("select e from Envelope e"
+        + " where e.jurisdiction = :jurisdiction"
+        + "   and e.status = 'UPLOAD_FAILURE'" // todo: use a constant
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepository.java]: enum can't be used, won't compile when used in annotation :(

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    public void findEnvelopesToResend_should_not_return_envelopes_that_failed_to_be_sent_too_many_times() throws Exception {
+        // given
+        final String jurisdiction = "X";
+        final int maxFailCount = 5;
+
+        dbHas(
+            envelopeWithFailureCount(maxFailCount - 2, jurisdiction),
+            envelopeWithFailureCount(maxFailCount - 1, jurisdiction),
+            envelopeWithFailureCount(maxFailCount, jurisdiction),
+            envelopeWithFailureCount(maxFailCount +1, jurisdiction),
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java]: space after + ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+@RunWith(SpringRunner.class)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest
+public class EnvelopeRepoTest {
+
+    @Autowired
+    private EnvelopeRepository repo;
+
+    @Test
+    public void findEnvelopesToResend_should_not_return_envelopes_that_failed_to_be_sent_too_many_times() throws Exception {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java]: findEnvelopesToResend_should_not_return_envelopes_that_failed_to_be_sent_too_many_times -> findEnvelopesToResend_should_not_return_envelopes_that_reached_max_fail_count ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    private Envelope envelopeWithFailureCount(int failCount, String jurisdiction) throws Exception {
+        Envelope envelope = envelope(jurisdiction, Status.UPLOAD_FAILURE);
+        envelope.setUploadFailureCount(failCount);
+        return envelope;
+    }
+
+    private void dbHas(Envelope... envelopes) {
+        Arrays.asList(envelopes).forEach(env -> repo.save(env));
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java]: repository has ability to `saveAll` entities at once. no need to foreach the iterable

Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepoTest.java]: done

**Pull request merged**